tests/integration/Core/Content/Category/SalesChannel/NavigationRouteTest.php

```diff
@@ -239,6 +242,188 @@ public function testInclude(): void
         static::assertArrayNotHasKey('id', $response[0]);
     }
 
+    public function testLandingPageInternalLinkHasSeoUrl(): void
+    {
+        $landingPageId = Uuid::randomHex();
+        $this->getContainer()->get('landing_page.repository')->create([
+            [
+                'id' => $landingPageId,
+                'name' => 'Test Landing Page',
+                'url' => 'test-landing-page',
+                'active' => true,
+                'salesChannels' => [
+                    ['id' => $this->ids->get('sales-channel')],
+                ],
+            ],
+        ], Context::createDefaultContext());
+
+        $this->getContainer()->get('category.repository')->update([
+            [
+                'id' => $this->ids->get('category3'),
+                'type' => CategoryDefinition::TYPE_LINK,
+                'linkType' => CategoryDefinition::LINK_TYPE_LANDING_PAGE,
+                'internalLink' => $landingPageId,
+            ],
+        ], Context::createDefaultContext());
+
+        $response = $this->requestFooterNavigationWithSeoUrls();
+
+        $originalUrl = null;
```

Comment:
the checks to original url can be removed as we compare now with the exact string we expect in both cases